src/app/zap-templates/zcl/data-model/chip/thread-network-diagnostics-cluster.xml

```diff
@@ -101,10 +101,10 @@ limitations under the License.
     <attribute side="server" code="0x07" define="NEIGHBOR_TABLE" type="ARRAY" entryType="NeighborTableStruct" length="254" writable="false" optional="false">NeighborTable</attribute>
     <attribute side="server" code="0x08" define="ROUTE_TABLE" type="ARRAY" entryType="RouteTableStruct" length="254" writable="false" optional="false">RouteTable</attribute>
     <attribute side="server" code="0x09" define="PARTITION_ID" type="int32u" min="0x00000000" max="0xFFFFFFFF" writable="false" isNullable="true" optional="false">PartitionId</attribute>
-    <attribute side="server" code="0x0A" define="WEIGHTING" type="int8u" min="0x00" max="0xFF" writable="false"  isNullable="true" optional="false">Weighting</attribute>
-    <attribute side="server" code="0x0B" define="DATA_VERSION" type="int8u" min="0x00" max="0xFF" writable="false"  isNullable="true" optional="false">DataVersion</attribute>
-    <attribute side="server" code="0x0C" define="STABLE_DATA_VERSION" type="int8u" min="0x00" max="0xFF" writable="false"  isNullable="true" optional="false">StableDataVersion</attribute>
-    <attribute side="server" code="0x0D" define="LEADER_ROUTER_ID" type="int8u" min="0x00" max="0xFF" writable="false"  isNullable="true" optional="false">LeaderRouterId</attribute>
+    <attribute side="server" code="0x0A" define="WEIGHTING" type="int16u" max="0xFF" writable="false"  isNullable="true" optional="false">Weighting</attribute>
```

Comment:
Do you have any insight as to why this was changed to uin16_t if the max stays 0xFF ? In Thread, those values are expressed as uint8_t.

Reply:
The reason is that ember uses the max value for nullable markers if the value is nullable.

Given that this is a nullable attribute, a uint8_t can only represent 0-254 and 0xFF is reserved for null. As a result we use uint16_t and reserve 0xFFFF for null.

Apparently this is a place where spec was changed to support internal details of the SDK (but nobody followed up after https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7487